### PR TITLE
Tabs Examples: Improve high contrast support and code quality

### PR DESCRIPTION
**Pull request by jongund**
Tab example updates include:

* Updating js code to use latest coding standards
* Updated keyboard focus styling for high contrast modes
* Updated visual affordance for selected tab in high contrast modes
* Included in examples when tabindex=0 should be used with `role=tabpanel`
* Removed delete tab feature
* Added additional tabs to increase the utility of Home and End key behavior

[tab example with automatic activation](https://raw.githack.com/w3c/aria-practices/tablist-update/examples/tabs/tabs-automatic.html)

[tab example with manual activation](https://raw.githack.com/w3c/aria-practices/tablist-update/examples/tabs/tabs-manual.html)

* [x] [Editorial review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#editorial_review) by @mcking65
* [x] [Editorial review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#editorial_review) by @carmacleod 
* [x] [Functional review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#functional_review) by @carmacleod 
* [x] [Visual design review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#design_review) during APG Group meeting
* [x] [Accessibility review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#accessibility_review) with JAWS/NVDA on Windows 10 by @carmacleod
* [x] [Accessibility review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#accessibility_review) with VO in Safari/Chrome on macOS by @carmacleod
* [x] [Accessibility review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#accessibility_review) with VO in Safari/Chrome on iOS/iPadOS by @carmacleod and @jongund (see https://github.com/w3c/aria-practices/pull/2194#issuecomment-1015956293)
* [x] [Code review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#code_review) by @jesdaigle or @howard-e 
* [x] [Test review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#test_review) by @jesdaigle or @howard-e

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/2194.html" title="Last updated on Mar 22, 2022, 4:23 PM UTC (ec80a02)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/2194/fa7c62b...ec80a02.html" title="Last updated on Mar 22, 2022, 4:23 PM UTC (ec80a02)">Diff</a>

### CODE REVIEW

File: examples/tabs/tabs-automatic.html [hunk 1]
```diff
@@ -68,10 +121,10 @@ <h2 id="ex_label">Example</h2>
 
       <section>
         <h2>Accessibility Features</h2>
-        <p>
-          To demonstrate the effects of deleting a tab, the third tab, labeled <q>Joke</q>,
-          can be deleted when it has focus by pressing <kbd>Delete</kbd>.
-        </p>
+        <ul>
+          <li>The selected <code>tab</code> has a 2 pixel border when it is selected to make the selected tab discernible from the other tabs to people wiith visual impairments using operating system high contrast modes.</li>
```
Comment (nschonni) [inline examples/tabs/tabs-automatic.html:125]:
```suggestion
          <li>The selected <code>tab</code> has a 2 pixel border when it is selected to make the selected tab discernible from the other tabs to people with visual impairments using operating system high contrast modes.</li>
```

File: examples/tabs/tabs-automatic.html [hunk 2]
```diff
@@ -68,10 +121,10 @@ <h2 id="ex_label">Example</h2>
...
+          <li>A <code>span</code> element is used to create a focus outline around the text to make the focus disernible to people with visual impairments using operating system high contrast modes.</li>
```
Comment (nschonni) [inline examples/tabs/tabs-automatic.html:126]:
```suggestion
          <li>A <code>span</code> element is used to create a focus outline around the text to make the focus discernible to people with visual impairments using operating system high contrast modes.</li>
```

Comment (jongund) [reply examples/tabs/tabs-automatic.html]: thanks, fixed it

File: examples/tabs/tabs-automatic.html [hunk 3]
```diff
@@ -68,10 +121,10 @@ <h2 id="ex_label">Example</h2>
 
       <section>
         <h2>Accessibility Features</h2>
-        <p>
-          To demonstrate the effects of deleting a tab, the third tab, labeled <q>Joke</q>,
-          can be deleted when it has focus by pressing <kbd>Delete</kbd>.
-        </p>
+        <ul>
+          <li>The selected <code>tab</code> has a 2 pixel border when it is selected to make the selected tab discernible from the other tabs to people with visual impairments using operating system high contrast modes.</li>
```
Comment (carmacleod) [inline examples/tabs/tabs-automatic.html:125]: This sentence is a bit long and there's a bit of redundancy. I think we can remove some words to make it easier to read.
```suggestion
          <li>The selected <code>tab</code> has a 2 pixel border to make it discernible from other tabs, for people with visual impairments using operating system high contrast modes.</li>
```

File: examples/tabs/tabs-manual.html [hunk 4]
```diff
@@ -68,10 +122,10 @@ <h2 id="ex_label">Example</h2>
 
       <section>
         <h2>Accessibility Features</h2>
-        <p>
-          To demonstrate the effects of deleting a tab, the third tab, labeled <q>Joke</q>,
-          can be deleted when it has focus by pressing <kbd>Delete</kbd>.
-        </p>
+        <ul>
+          <li>The selected <code>tab</code> has a 2 pixel border when it is selected to make the selected tab discernible from the other tabs to people with visual impairments using operating system high contrast modes.</li>
```
Comment (carmacleod) [inline examples/tabs/tabs-manual.html:126]: Same as above, removing some words may make this sentence easier to read.

```suggestion
          <li>The selected <code>tab</code> has a 2 pixel border to make it discernible from other tabs, for people with visual impairments using operating system high contrast modes.</li>
```

File: examples/tabs/js/tabs-automatic.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  /* EVENT HANDLERS */
+
+  handleKeydown(event) {
+    var tgt = event.currentTarget,
+      flag = false;
+
+    switch (event.key) {
+      case 'ArrowUp':
```
Comment (carmacleod) [inline examples/tabs/js/tabs-automatic.js:94]: I don't think Up and Down arrow should be used to navigate through a horizontal tab list. Users prefer to keep Up and Down for scrolling the page.

If it was a vertical tab list, then Up and Down would have to be used.

File: examples/tabs/js/tabs-automatic.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      case 'ArrowLeft':
+        this.setSelectedToPreviousTab(tgt);
+        flag = true;
+        break;
+
+      case 'ArrowDown':
```
Comment (carmacleod) [inline examples/tabs/js/tabs-automatic.js:100]: As above, I think we need to allow the default page scroll action for Up/Down arrow.

File: examples/tabs/js/tabs-manual.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  /* EVENT HANDLERS */
+
+  handleKeydown(event) {
+    var tgt = event.currentTarget,
+      flag = false;
+
+    switch (event.key) {
+      case 'ArrowUp':
```
Comment (carmacleod) [inline examples/tabs/js/tabs-manual.js:92]:
```suggestion
```

File: examples/tabs/js/tabs-manual.js [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+      case 'ArrowLeft':
+        this.moveFocusToPreviousTab(tgt);
+        flag = true;
+        break;
+
+      case 'ArrowDown':
```
Comment (carmacleod) [inline examples/tabs/js/tabs-manual.js:98]:
```suggestion
```

**Review by carmacleod — COMMENTED** [review]
@jongund This is looking really good! I did functional testing and editorial testing. Have not done accessibility testing, but I will. I think it's important not to support Up and Down arrow keys in a horizontal tablist. Users like to use Up/Down arrows to scroll the page, and it can be disorienting if those keys are taken by a horizontal tab control.

File: examples/tabs/tabs-automatic.html (hunk 3 above)
Comment (jongund) [reply examples/tabs/tabs-automatic.html:130]: Fixed

File: examples/tabs/tabs-manual.html (hunk 4 above)
Comment (jongund) [reply examples/tabs/tabs-manual.html:127]: fixed

File: examples/tabs/js/tabs-automatic.js (hunk 5 above)
Comment (jongund) [reply examples/tabs/js/tabs-automatic.js:94]: fixed

File: examples/tabs/js/tabs-automatic.js (hunk 6 above)
Comment (jongund) [reply examples/tabs/js/tabs-automatic.js:100]: fixed

File: examples/tabs/js/tabs-manual.js (hunk 7 above)
Comment (jongund) [reply examples/tabs/js/tabs-manual.js]: fixed

File: examples/tabs/js/tabs-manual.js (hunk 8 above)
Comment (jongund) [reply examples/tabs/js/tabs-manual.js]: fixed

File: examples/tabs/js/tabs-automatic.js (hunk 6 above)
Comment (jongund) [reply examples/tabs/js/tabs-automatic.js:100]: fixed

File: examples/tabs/js/tabs-automatic.js (hunk 5 above)
Comment (jongund) [reply examples/tabs/js/tabs-automatic.js:94]: fixed

File: examples/tabs/tabs-manual.html [hunk 9]
```diff
@@ -68,10 +122,11 @@ <h2 id="ex_label">Example</h2>
 
       <section>
         <h2>Accessibility Features</h2>
-        <p>
-          To demonstrate the effects of deleting a tab, the third tab, labeled <q>Joke</q>,
-          can be deleted when it has focus by pressing <kbd>Delete</kbd>.
-        </p>
+        <ul>
+          <li>Since the first focusable element in each <code>tabpanel</code> is a focusable element (e.g. link), the tabpanel does <strong>not</strong> need to set a <code>tabindex</code> value to include it in the tab sequence of the page.</li>
```
Comment (carmacleod) [inline examples/tabs/tabs-manual.html:126]: Just removing redundant word "focusable".

```suggestion
          <li>Since the first element in each <code>tabpanel</code> is a focusable element (e.g. link), the tabpanel does <strong>not</strong> need to set a <code>tabindex</code> value to include it in the tab sequence of the page.</li>
```

**Comment by a11ydoer** [comment]
Thanks so much for assigning yourself for the review, @carmacleod

File: examples/tabs/css/tabs.css [hunk 10]
```diff
@@ -1,107 +1,63 @@
...
-  right: 0;
-  left: 0;
-  height: 0.3em;
+  padding: 2px 2px 4px;
+  border-width: 2px;
+  border-bottom-color: hsl(220deg 43% 99%);
   background: hsl(220deg 43% 99%);
-  box-shadow: none;
-  content: "";
 }
```
Comment (carmacleod) [inline examples/tabs/css/tabs.css:32]: Unselected tabs look like they are drawn 1px below the bottom of the selected tab (or the top of the tabpanel).
I think there needs to be one more rule that says that unselected tabs only have a 1px bottom border instead of 2px:
```suggestion
[role="tab"][aria-selected="false"] {
    border-bottom: 1px solid hsl(219deg 1% 72%);
}
```

Before new rule:
image


After new rule:
image

**Comment by carmacleod** [comment]
Completed review of JAWS/NVDA in Chrome/FF on Windows 10, and all works well with one exception:
JAWS does not say anything when the user tabs into a tabpanel that has tabindex="0", as seen in the [](https://raw.githack.com/w3c/aria-practices/tablist-update/examples/tabs/tabs-automatic.html). NVDA works fine.
I will open an issue with JAWS as soon as we merge this PR.

**Comment by carmacleod** [comment]
Completed review of VO in Safari/Chrome on macOS. Things work differently between the two browsers, however the tabs control is quite usable in both Safari and Chrome for both examples. One oddity is that in Safari, VO says "main, [tabpanel label], tabpanel" when the user tabs into a tabpanel with tabindex="0" in the automatic activation example.

**Comment by carmacleod** [comment]
Tried reviewing VO in Safari on iOS and iPadOS using gestures. Both Tabs controls are useable, but they both need swipe and double-tap to select the tab, even for the automatic activation example, so they are both behaving like manual activation on desktop. This is probably because there's no gesture (that I know of) for arrow keys. It works ok, though.

A problem that makes gestures suboptimal in a Tabs control is that you have to swipe through all of the tabs before you get to the selected tabpanel. This is because there's no gesture for tab key either. Swipe just moves the point of regard. So even though swiping through 4 tabs before getting to the tabpanel in this example is easy enough, it would be pretty bad if there were a dozen tabs or more. I have no idea what to do about this. Does anyone know of a better way? I can't find anything helpful in the rotor or in the [list of gestures](https://support.apple.com/en-ca/guide/ipad/ipad9a246584/ipados).

NOTE: I would really like someone else to try this on iOS and iPadOS and report back here. Particularly on iOS, as I was experiencing a lot of user error, where I seemed to be moving the point of regard to random places on the screen of my old iPhone SE even though I was being very careful not to accidentally use explore by touch when I was swiping.

**Comment by JAWS-test** [comment]
> JAWS does not say anything when the user tabs into a tabpanel that has tabindex="0"

See: https://github.com/FreedomScientific/VFO-standards-support/issues/527, 
https://github.com/FreedomScientific/VFO-standards-support/issues/542, https://github.com/FreedomScientific/VFO-standards-support/issues/519#issuecomment-827550108, https://github.com/FreedomScientific/VFO-standards-support/issues/151, https://github.com/FreedomScientific/VFO-standards-support/issues/316, https://github.com/FreedomScientific/VFO-standards-support/issues/491

**Comment by carmacleod** [comment]
Thank-you, @JAWS-test. Looks like there are a lot of similar tabpanel problems with JAWS. :)
I will read through them all, and maybe try to tie them together, after this PR is merged.

Please note that this PR changes the example URLs, so after this is merged, the following links will no longer work:
- https://w3c.github.io/aria-practices/examples/tabs/tabs-1/tabs.html
- https://w3c.github.io/aria-practices/examples/tabs/tabs-2/tabs.html

Instead, the JAWS issues will need to use these URLs:
- https://raw.githack.com/w3c/aria-practices/tablist-update/examples/tabs/tabs-automatic.html
- https://raw.githack.com/w3c/aria-practices/tablist-update/examples/tabs/tabs-manual.html

**Review by carmacleod — APPROVED** [review]
This looks good to me now, and I am approving this PR for Editorial, Functional, and Accessibility on desktop. Thanks, Jon!

Still needs Code and Test review before merging, and I would like someone else to review Accessibility on touch/mobile please. Did we review visual design on the call?

**Comment by a11ydoer** [comment]
@carmacleod We discussed the PR today's meeting. Editorial, code, and test review are the only ones left before we close the PR. Thanks for all your review work.

**Comment by jongund** [comment]
@jongund 
I tried the examples with VO on iOS and it works, but you have to swipe through un selected tabs to get to the selected tab panel.  Not sure if there is anything our example can do about that, seems to a be a feature AT would need to implement.

**Comment by a11ydoer** [comment]
tagging @cookiecrook with curiosity on the issue. 

> @jongund I tried the examples with VO on iOS and it works, but you have to swipe through un selected tabs to get to the selected tab panel. Not sure if there is anything our example can do about that, seems to a be a feature AT would need to implement.

**Comment by jongund** [comment]
@jesdaigle or @howard-e Can one of you assign your self as a reviewer and do the code review today for this pull request?   

The code is not very complicated and the regression tests are a little simpler than the tests fir the examples they are replacing.   

Thank you for your help.

**Comment by jesdaigle** [comment]
> 

@jongund I am unable to assign it to myself but I did review.

**Comment by jongund** [comment]
@jesdaigle 
Thank you!

**Comment by jongund** [comment]
@jesdaigle 
I was able to add you as a reviewer.

**Review by jesdaigle — APPROVED** [review]

**Review by mcking65 — CHANGES_REQUESTED** [review]
I'd like to discuss the use of the word "selected" in the example name.
I think we also need to add more explanation to the high contrast support and make the wording more consistent with other examples.

File: examples/tabs/tabs-automatic.html [hunk 11]
```diff
@@ -68,10 +125,11 @@ <h2 id="ex_label">Example</h2>
...
       <section>
         <h2>Accessibility Features</h2>
-        <p>
-          To demonstrate the effects of deleting a tab, the third tab, labeled <q>Joke</q>,
-          can be deleted when it has focus by pressing <kbd>Delete</kbd>.
-        </p>
+        <ul>
+          <li>Since the tabpanels do not contain any focusable elements, <code>tabindex="0"</code> is used to include the tabpanel in the tab sequence of the page.</li>
+          <li>The selected <code>tab</code> has a 2 pixel border to make it discernible from other tabs, for people with visual impairments using operating system high contrast modes.</li>
+          <li>A <code>span</code> element is used to create a focus outline around the text to make the focus discernible to people with visual impairments using operating system high contrast modes.</li>
```
[the remark this answers is not in the document]
Comment (a11ydoer) [reply examples/tabs/tabs-automatic.html:131]: resolved with APG meeting on 2/22/2022 - Jon will update the pr.

File: examples/tabs/tabs-automatic.html [hunk 12]
```diff
@@ -68,10 +125,11 @@ <h2 id="ex_label">Example</h2>
 
       <section>
         <h2>Accessibility Features</h2>
-        <p>
-          To demonstrate the effects of deleting a tab, the third tab, labeled <q>Joke</q>,
-          can be deleted when it has focus by pressing <kbd>Delete</kbd>.
-        </p>
+        <ul>
+          <li>Since the tabpanels do not contain any focusable elements, <code>tabindex="0"</code> is used to include the tabpanel in the tab sequence of the page.</li>
+          <li>The selected <code>tab</code> has a 2 pixel border to make it discernible from other tabs, for people with visual impairments using operating system high contrast modes.</li>
```
[the remark this answers is not in the document]
Comment (jongund) [reply examples/tabs/tabs-automatic.html:130]: @mcking65 
Made some updates for your review.

File: examples/tabs/tabs-automatic.html (hunk 11 above)
[the remark this answers is not in the document]
Comment (jongund) [reply examples/tabs/tabs-automatic.html:131]: @mcking65 
Made some updates on the Accessibility Features for your review

File: aria-practices.html [hunk 13]
```diff
@@ -2772,8 +2772,8 @@ <h3>Tabs</h3>
       <section class="notoc">
         <h4>Examples</h4>
         <ul>
-          <li><a href="examples/tabs/tabs-1/tabs.html">Tabs With Automatic Activation</a>: A tabs widget where tabs are automatically activated and their panel is displayed when they receive focus.</li>
-          <li><a href="examples/tabs/tabs-2/tabs.html">Tabs With Manual Activation</a>: A tabs widget where users activate a tab and display its panel by pressing <kbd>Space</kbd> or <kbd>Enter</kbd>.</li>
+          <li><a href="examples/tabs/tabs-automatic.html">Tabs With Automatic Activation</a>: A tabs widget where tabs are automatically activated and their panel is displayed when they receive focus.</li>
```
Comment (ZoeBijl) [inline aria-practices.html:2775]: This is a lot clearer, nice!

File: aria-practices.html [hunk 14]
```diff
@@ -2835,6 +2835,7 @@ <h4>Keyboard Interaction</h4>
             </ol>
           </li>
           <li>If the tab list is horizontal, it does not listen for <kbd>Down Arrow</kbd> or <kbd>Up Arrow</kbd> so those keys can provide their normal browser scrolling functions even when focus is inside the tab list.</li>
+          <li>When the tabpanel does not contain any focusable elements or the first element with content is not focusable, the tabpanel should set <code>tabindex="0"</code> to include it in the tab sequence of the page.</li>
```
Comment (ZoeBijl) [inline aria-practices.html:2838]: > or the first element with content is not focusable

What does that mean? Could you please elaborate?

File: examples/tabs/css/tabs.css [hunk 15]
```diff
@@ -1,107 +1,67 @@
 .tabs {
-  width: 20em;
   font-family: "lucida grande", sans-serif;
 }
 
 [role="tablist"] {
-  margin: 0 0 -0.1em;
-  overflow: visible;
+  min-width: 550px;
```
Comment (ZoeBijl) [inline examples/tabs/css/tabs.css:6]: Why are we setting a min-width?

File: examples/tabs/css/tabs.css [hunk 16]
```diff
@@ -1,107 +1,67 @@
...
-  content: "";
+  outline: none;
+  font-weight: bold;
 }
 
 [role="tab"][aria-selected="true"] {
-  border-radius: 0;
+  padding: 2px 2px 4px;
+  border-width: 2px;
+  border-bottom-color: hsl(220deg 43% 99%);
```
Comment (ZoeBijl) [inline examples/tabs/css/tabs.css:29]: I don't understand any of these CSS changes? They don't seem to relate to the requested changes?

File: examples/tabs/css/tabs.css [hunk 17]
```diff
@@ -1,107 +1,67 @@
...
-[role="tab"]:focus::before,
-[role="tab"][aria-selected="true"]::before {
-  position: absolute;
-  bottom: 100%;
-  right: -1px;
-  left: -1px;
-  border-radius: 0.2em 0.2em 0 0;
-  border-top: 3px solid hsl(20deg 96% 48%);
-  content: "";
+  outline: none;
```
Comment (ZoeBijl) [inline examples/tabs/css/tabs.css:22]: We should probably set a transparent outline.

File: examples/tabs/css/tabs.css [hunk 18]
```diff
@@ -1,107 +1,67 @@
...
-  border: 1px solid hsl(219deg 1% 72%);
-  border-radius: 0 0.2em 0.2em;
-  box-shadow: 0 0 0.2em hsl(219deg 1% 72%);
+  padding: 5px;
+  border: 2px solid hsl(219deg 1% 72%);
+  border-radius: 0 5px 5px;
   background: hsl(220deg 43% 99%);
+  min-height: 10em;
+  min-width: 550px;
+  overflow: auto;
```
Comment (ZoeBijl) [inline examples/tabs/css/tabs.css:58]: Same here, I don't understand why these changes are needed. Can you elaborate?

File: examples/tabs/js/tabs-automatic.js [hunk 19]
```diff
@@ -0,0 +1,136 @@
+/*
```
Comment (ZoeBijl) [inline examples/tabs/js/tabs-automatic.js:1]: Did you make any changes to the JS? The delete/add makes it impossible to tell.

**Review by ZoeBijl — CHANGES_REQUESTED** [review]
I don't quite get the reasoning behind a lot of these changes. In addition I think the design changes as they stand are a step backwards.

File: aria-practices.html (hunk 14 above)
Comment (jongund) [reply aria-practices.html:2838]: There needs to be more clarity when `tabindex=0` is put on the element with the `tabpanel` role.  So the modified examples now show one example where the tabindex is on the tabpanel and one that it is not based on the requirements of the first element being focusable in the tabpanel.

File: examples/tabs/css/tabs.css (hunk 16 above)
Comment (jongund) [reply examples/tabs/css/tabs.css:29]: To support high contrast modes I am trying to emphasize at least 2 pixel spacing and borders to support operating system high contrast modes for focus styling.  This is easier for developers to see if pixels are used in the code.

File: examples/tabs/css/tabs.css (hunk 17 above)
Comment (jongund) [reply examples/tabs/css/tabs.css:22]: The updated examples are actively styling keyboard focus, so it will be visually confusing if outline also appears.  We do not want two focus rings.

File: aria-practices.html (hunk 13 above)
Comment (jongund) [reply aria-practices.html:2775]: Two main issues:
1.  I am not sure it is a a good practice for the APG to reference contemporary musicians, it might be misunderstood as some type of endorsement for those musicians or at least an unfair improvement to their SEO, so I did not feel comfortable using them in the updates I made. 
2. We will be adding at least one more example that will add and delete tabs and this was easier for me to find additional related content for those examples.

File: examples/tabs/tabs-manual.html [hunk 20]
```diff
@@ -43,23 +46,74 @@ <h2 id="ex_label">Example</h2>
...
+              aria-labelledby="tab-3"
+              class="is-hidden">
+              <p><a href="https://en.wikipedia.org/wiki/Ida_Henriette_da_Fonseca">Ida Henriette da Fonseca</a> (July 27, 1802 – July 6, 1858) was a Danish opera singer and composer.  Ida Henriette da Fonseca was the daughter of Abraham da Fonseca (1776–1849) and Marie Sofie Kiærskou (1784–1863). She and her sister Emilie da Fonseca were students of Giuseppe Siboni, choir master of the Opera in Copenhagen. She was given a place at the royal Opera alongside her sister the same year she debuted in 1827.</p>
+            </div>
+            <div
+              id="tabpanel-4"
+              role="tabpanel"
+              aria-labelledby="tab-4"
+              class="is-hidden">
+              <p><a href="https://en.wikipedia.org/wiki/Peter_Lange-M%C3%BCller">Peter Erasmus Lange-Müller</a> (1 December 1850 – 26 February 1926) was a Danish composer and pianist. His compositional style was influenced by Danish folk music and by the work of Robert Schumann; Johannes Brahms; and his Danish countrymen, including J.P.E. Hartmann.</p>
```
[the remark this answers is not in the document]
Comment (jongund) [reply examples/tabs/tabs-manual.html:116]: Two main issues:
1.  I am not sure it is a a good practice for the APG to reference contemporary musicians, it might be misunderstood as some type of endorsement for those musicians or at least an unfair improvement to their SEO, so I did not feel comfortable using them in the updates I made. 
2. We will be adding at least one more example that will add and delete tabs and this was easier for me to find additional related content for those examples.

File: aria-practices.html (hunk 13 above)
Comment (ZoeBijl) [reply aria-practices.html:2775]: Should we also update the carousel example to remove all references to the TV shows?

File: examples/tabs/css/tabs.css [hunk 21]
```diff
@@ -1,107 +1,67 @@
...
-[role="tab"] {
+[role="tab"],
+[role="tab"]:focus,
+[role="tab"]:hover {
   position: relative;
+  z-index: 2;
+  top: 2px;
   margin: 0;
-  padding: 0.3em 0.5em 0.4em;
+  padding: 3px 3px 4px;
```
[the remark this answers is not in the document]
Comment (ZoeBijl) [reply examples/tabs/css/tabs.css:16]: How does changing this to `px` help show there’s a two pixel border?

**Comment by cookiecrook** [comment]
> tagging @cookiecrook with curiosity on the issue.
> 
> > @jongund I tried the examples with VO on iOS and it works, but you have to swipe through un selected tabs to get to the selected tab panel. 

Since iOS doesn’t have a UI concept similar to “managed focus” on a tab group, I think this is expected behavior.

File: examples/tabs/css/tabs.css (hunk 16 above)
Comment (ZoeBijl) [reply examples/tabs/css/tabs.css:29]: Ah I see what you’re saying now. But wouldn’t a transparent outline achieve the same for high contrast mode? Would’ve been a far less invasive change.

**Comment by jongund** [comment]
@ZoeBijl 
Transparent borders or outlines cause problems in high contrast modes in Windows.  Outline is also used primarily for keyboard focus and in this case the focus border is put on a span inside the tab with focus to make it more visible to people with visual impairments.

File: examples/tabs/css/tabs.css (hunk 18 above)
Comment (ZoeBijl) [reply examples/tabs/css/tabs.css:58]: Is this related to the content change? Can you elaborate?

**Comment by ZoeBijl** [comment]
> @ZoeBijl Transparent borders or outlines cause problems in high contrast modes in Windows. Outline is also used primarily for keyboard focus and in this case the focus border is put on a span inside the tab with focus to make it more visible to people with visual impairments.

I was under the impression that transparent outlines were a recommended way to support high contrast mode? @smhigley and @melanierichards might have insights there.

**Comment by jongund** [comment]
@ZoeBijl 
In this example the focus ring is being put on a child span element of the tab element with focus to make the focus ring separate from the border of the tab element to make it easier for people with visual impairments identify which tab has focus.   Outline is set to zero on the tab with focus so there are not two focus rings when the tab gets focus.   Using the child span element allows for the 2 pixels of spacing between the border visually identifying the element as a tab and the span being used to show the tab has keyboard focus.

**Comment by mcking65** [comment]
@jongund 

In radio group, we used the following language related to transparency:

>Because transparent borders are visible on some systems with high contrast enabled, only the focused element has a visible border. When focusable elements are not focused, they have a 0-width border and additional padding equal in width to the border that is used to indicate focus.

Is some form of this useful here?

**Comment by mcking65** [comment]
@jongund 

Would copy something like the following work for accessibility features? I want to be sure we are clearly communicating the purpose of the important design and coding decisions.

>To ensure people who rely on browser or operating system high contrast settings can both distinguish the active (selected) tab from other tabs and perceive keyboard focus:
>* The active tab  has a 2 pixel border on its left, top, and right sides while the names of inactive tabs do not have a border.
>* The focus ring is drawn with CSS focus outline on a child span of the tab element. This focus span is separated from the tab border by 2 pixels of space to ensure focus and selection are separately perceivable. Note that when a tab element is focused, the outline of the tab element itself is set to 0 so that only one focus ring is displayed.

**Comment by jongund** [comment]
@mcking65 
I updated the accessibility feature information based on your feedback, please review when you have a chance.

**Comment by jnurthen** [comment]
The selected tab. is shown differently at the initial page load vs after an explicit selection has been made when running in WHCM
Before explicit selection showing a slightly thicker border in the white color for selection
After explicit selection showing a slightly thicker border in the blue Highlight color for selection

**Comment by jongund** [comment]
@jnurthen 
I have updated the examples to make the active tab 4 pixels higher to make it easier to discern in high contrast mode.   What do you think?

**Comment by jongund** [comment]
@mcking65 @jnurthen 
Your changes in documentation look good to me.  The change in the top border width of the active tab should also satisfy James's  report of the active tab being easily identifiable in high contrast modes.

**Comment by jongund** [comment]
@mcking65 
Updated the heading to use H3 instead of h2.

**Comment by ZoeBijl** [comment]
@mcking65 I’ll re-review with a fresh pair of eyes. While I might not agree with the visual changes; that’s more a personal thing than an APG thing so I’ll let that go. Should be done this week.

File: examples/tabs/js/tabs-automatic.js (hunk 19 above)
Comment (ZoeBijl) [reply examples/tabs/js/tabs-automatic.js:1]: Can you elaborate?
